client/client.go

```diff
@@ -660,3 +775,15 @@ func (rq *defaultRequester) getWebsocket(urlPath string) (clientWebsocket, error
 	}
 	return conn, err
 }
+
+// getIdentityFingerprint extracts the public key from the certificate and
+// calculates the fingerprint.
+func getIdentityFingerprint(cert *x509.Certificate) (string, error) {
+	pubKey, ok := cert.PublicKey.(ed25519.PublicKey)
+	if !ok {
+		return "", errors.New("certificate must use Ed25519 public key")
+	}
+	hashBytes := sha512.Sum384(pubKey)
```

Comment:
Is this always SHA-384?

Reply:
This was chosen as per earlier specifications for now. I think there is room to increase this at a later point, easily.

Reply:
Discussed, resolving.